pkg/server/router/presentation.go

```diff
@@ -0,0 +1,140 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"github.com/TBD54566975/ssi-sdk/credential/exchange"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/tbd54566975/ssi-service/pkg/server/framework"
+	svcframework "github.com/tbd54566975/ssi-service/pkg/service/framework"
+	"github.com/tbd54566975/ssi-service/pkg/service/presentation"
+	"net/http"
+)
+
+type PresentationDefinitionRouter struct {
+	service *presentation.Service
+}
+
+func NewPresentationDefinitionRouter(s svcframework.Service) (*PresentationDefinitionRouter, error) {
+	if s == nil {
+		return nil, errors.New("service cannot be nil")
+	}
+	service, ok := s.(*presentation.Service)
+	if !ok {
+		return nil, fmt.Errorf("could not create presentation router with service type: %s", s.Type())
+	}
+	return &PresentationDefinitionRouter{service: service}, nil
+}
+
+type CreatePresentationDefinitionRequest struct {
+	PresentationDefinition exchange.PresentationDefinition `json:"presentation_definition" validate:"required"`
```

Comment:
we have been doing camelCase
Change the json strings to presentationDefinition

Reply:
I do not think we should take in an entire presentation definition here, instead we should take in what is necessary to build one. this is similar to our manifest and credential apis today:

see: https://github.com/TBD54566975/ssi-service/blob/main/pkg/server/router/manifest.go#L43

Reply:
For the UI this may push some logic into their code to build a compliant presentation definition instead of passing in the form fields and us building the object ourselves. Another factor at play is if we use the `Builder` methods we have in the sdk for each of these objects we can have higher certainty that we're using the latest supported object models and always building spec-compliant objects.

I can definitely see the appeal of the approach you presented - just have to consider the tradeoffs.

One other note - with an optional `keyID` we would need to create our own ID to map to theirs or else risk ID collisions.